src/Message.chpl

```diff
@@ -35,13 +36,274 @@ module Message {
+            return str2dtype(this.dtype);
+        }
+
+        /*
+        * Return the raw string value
+        * Returns string
+        */
+        proc getValue() {
+            return this.val;
+        }
+
+        /*
+        * Return the value as int64
+        * Returns int
+        */
+        proc getIntValue(): int throws {
+            try {
+                return this.val:int;
+            }
+            catch {
+                throw new owned ErrorWithContext("Parameter cannot be cast as int. Attempting to cast %s as type int failed".format(this.val),
+                                    getLineNumber(),
+                                    getRoutineName(),
+                                    getModuleName(),
+                                    "TypeError");
+            }
+        }
+
+        /*
+        * Return the value as uint64
+        * Returns uint
+        */
+        proc getUIntValue(): uint throws {
+            try {
+                return this.val:uint;
+            }
+            catch {
+                throw new owned ErrorWithContext("Parameter cannot be cast as uint. Attempting to cast %s as type uint failed".format(this.val),
+                                    getLineNumber(),
+                                    getRoutineName(),
+                                    getModuleName(),
+                                    "TypeError");
+            }
+        }
+
+        /*
+        * Return the value as float64
+        * Returns real
+        */
+        proc getRealValue(): real throws {
+            try {
+                return this.val:real;
+            }
+            catch {
+                throw new owned ErrorWithContext("Parameter cannot be cast as real. Attempting to cast %s as type real failed".format(this.val),
+                                    getLineNumber(),
+                                    getRoutineName(),
+                                    getModuleName(),
+                                    "TypeError");
+            }
+        }
+
+        /*
+        * Return the value as bool
+        * Returns bool
+        */
+        proc getBoolValue(): bool throws {
+            try {
+                return this.val:bool;
+            }
+            catch {
+                throw new owned ErrorWithContext("Parameter cannot be cast as bool. Attempting to cast %s as type bool failed".format(this.val),
+                                    getLineNumber(),
+                                    getRoutineName(),
+                                    getModuleName(),
+                                    "TypeError");
+            }
+        }
+
+        /*
+        * Return the value as the provided type
+        */
+        proc getValueAsType(type t = string): t throws {
+            if objType != dtype {
+                throw new owned ErrorWithContext("The value provided is not a castable type, please use ParameterObj.getSymEntry for this object.".format(t, this.val, t),
```

Comment:
this has stuff in the `.format` but no `%t/%s` or other format strings for them to go. Looks like it's leftover from copying the one below

Reply:
Corrected